src/nvim/window.c

```diff
@@ -213,7 +213,7 @@ void do_window(int nchar, int Prenum, int xchar)
 
 #define CHECK_CMDWIN \
   do { \
-    if (cmdwin_type != 0) { \
+    if (curwin == cmdwin_win && cmdwin_lock) { \
```

Comment:
Likewise, this allows entering other windows once in the messages window with `<C-W>w` and the like.